packages/core/core-flows/src/inventory/steps/create-inventory-items.ts

```diff
@@ -1,31 +1,50 @@
-import { StepResponse, createStep } from "@medusajs/workflows-sdk"
-
 import { ModuleRegistrationName } from "@medusajs/modules-sdk"
-import { IInventoryServiceNext, InventoryNext } from "@medusajs/types"
+import {
+  CreateVariantInventoryItemDTO,
+  IInventoryServiceNext,
+  InventoryNext,
+} from "@medusajs/types"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
 
 export const createInventoryItemsStepId = "create-inventory-items"
 export const createInventoryItemsStep = createStep(
   createInventoryItemsStepId,
-  async (data: InventoryNext.CreateInventoryItemInput[], { container }) => {
+  async (data: CreateVariantInventoryItemDTO[], { container }) => {
     const inventoryService: IInventoryServiceNext = container.resolve(
       ModuleRegistrationName.INVENTORY
     )
+    const variantItemDataMap = new Map<
+      string,
+      InventoryNext.CreateInventoryItemInput[]
+    >()
+    const variantItemsMap = new Map<string, InventoryNext.InventoryItemDTO[]>()
+
+    for (const { variant_id, ...inventoryData } of data) {
+      const array = variantItemDataMap.get(variant_id) || []
+
+      array.push(inventoryData)
+      variantItemDataMap.set(variant_id, array)
+    }
+
+    for (const [variantId, itemsData] of variantItemDataMap.entries()) {
+      const createdItems = await inventoryService.create(itemsData)
```

Comment:
don't we have this create in bulk on the module?
that can be a problem if you have multiple and one of them fail, the compensation will have no id.
so, worst case scenario would be bring that loop to the module, so you can reuse the same db transaction

Reply:
I'll change this to be a purely create item workflow and create a different workflow that is to attach an items to a variant.

Reply:
usually in cases like this we must guarantee the order we sent to be created is the same the method will return, so we can remap it